### ISSUE TITLE
ERROR Error: Uncaught (in promise): TypeError: r is not a constructor

### ISSUE THREAD

**Issue by abdatta** [bruckwubete/angular-2-carousel-3d#1]
I am getting this error when I do `ng --prod` or `ng --prod --aot`.
But it runs perfectly during development when I do `ng serve`.

![error](https://user-images.githubusercontent.com/20218826/34514221-5a3854d4-f092-11e7-9818-45e5743d0c78.png)

I found a similar error in some other library [here](https://github.com/jvandemo/generator-angular2-library/issues/229). Please look into the error at the earliest as I am looking forward to use this beautiful carousal.

**Comment by bruckwubete** [comment]
@abdatta I cannot reproduce this.

**Comment by abdatta** [comment]
@bruckwubete I installed this repo from npm into my Angular app and used it simply as it is mentioned on the npm page. Everything worked perfectly fine on my development server (that is, when I do `ng serve`). However when I tried to build it for production (using `ng build --prod` or `ng build --prod --aot`) then it initially builds successfully without showing any errors, but when I open this build statically (by serving them using some other server like Apache or Nginx), then it doesn't open and instead shows the error(as in the screenshot) on the developer console. However doing simply `ng build` doesn't cause this problem. Currently I am serving my app using the development build only, waiting for you to find a way to get this fixed.
